python/TestHarness/TestHarness.py

```diff
@@ -235,87 +242,140 @@ def launchTest(self, find_only, file, dirpath, testers):
+                                command = 'qstat -x -f %s' % (pbs_job['id'])
+                            else:
+                                continue
+
+                        # Create qsub script.
+                        else:
+                            # Override command with the qsub command needed to launch a PBS job
+                            command = self.createQsubFile(command, dirpath, tester)
+
+                    # This method spawns another process and allows this loop to continue looking for tests
+                    # RunParallel will call self.testOutputAndFinish when the test has completed running
+                    # This method will block when the maximum allowed parallel processes are running
+                    if self.options.dry_run:
                         self.handleTestStatus(tester)
-                self.runner.jobSkipped(tester.parameters()['test_name'])
+                    else:
+                        self.runner.run(tester, command)
+                else: # This job is skipped - notify the runner
+                    status = tester.getStatus()
+                    if status != tester.bucket_silent: # SILENT occurs when a user is using --re options
+                        if (self.options.report_skipped and status == tester.bucket_skip) \
+                           or status == tester.bucket_skip:
+                            self.handleTestStatus(tester)
+                        elif status == tester.bucket_deleted and self.options.extra_info:
+                            self.handleTestStatus(tester)
+                    self.runner.jobSkipped(tester.parameters()['test_name'])
+
+    # Create the PBS qsub script, perform directory/file operations
+    # TODO: Because we need to open this file a second time to modify
+    # any would-be prereq attribute, should we refactor this, and move
+    # this routine to RunParallel (if possible)?
+    # see addJobID in RunParallel for clarification.
+    def createQsubFile(self, command, dirpath, tester):
+        pbs_job_dir = os.path.join(dirpath, 'job_' + self.options.pbs)
+        if not os.path.exists(pbs_job_dir):
+            os.mkdir(pbs_job_dir)
+        os.chdir(pbs_job_dir)
+        job_instance = self.factory.create('PBSJob', tester.specs['test_name'], tester.specs)
+        specs = job_instance.specs
+        path_friendly = specs['test_name'].replace('/', '_')
+
+        ## Set some PBS specific specs that help populate the qsub script (and directory)
+        specs['job_name'] = path_friendly
+        specs['test_dir'] = pbs_job_dir
+        specs['template_script'] = self.moose_dir + '/python/TestHarness/pbs_submit.sh'
+        specs['no_copy_pattern'] = '.*\.sh'
+        specs['copy_files'] = 'gold'
+        specs['mpi_procs'] = '1'
+        specs['soft_link_output'] = False
+        specs['place'] = 'scatter:excl'
+        specs['chunks'] = '1'
+        specs['no_copy'] = self.options.input_file_name
```

Comment:
Hmm... I'm not entirely sure I'm sold on populating the parameters like this. I wonder if we should built another subclass of PBSJob that defaults the parameters to these values? Just thinking out loud here.